src/common/src/cast/mod.rs

```diff
@@ -160,6 +176,40 @@ pub fn i64_to_timestamp(t: i64) -> Result<Timestamp> {
     ))
 }
 
+/// Converts str to interval for postgres debezium cdc json
+///
+/// The input str must have the following format:
+/// P<years>Y<months>M<days>DT<hours>H<minutes>M<seconds>S
+///
+/// Example
+/// - P1Y2M3DT4H5M6.78S
+#[inline]
+pub fn str_to_interval(s: &str) -> Result<Interval> {
+    let re = Regex::new(r"P([0-9]+)Y([0-9]+)M([0-9]+)DT([0-9]+)H([0-9]+)M([0-9]+(?:\.[0-9]+)?)S")
+        .unwrap();
+    let caps = re
+        .captures(s)
+        .ok_or(PARSE_ERROR_STR_TO_INTERVAL.to_string())?;
+    // safe to unwrap, since capture group is `Some`
+    let years: i32 = caps.get(1).unwrap().as_str().parse().unwrap();
+    let months: i32 = caps.get(2).unwrap().as_str().parse().unwrap();
+    let days = caps.get(3).unwrap().as_str().parse().unwrap();
+    let hours: i64 = caps.get(4).unwrap().as_str().parse().unwrap();
```

Comment:
Does the source value include zero fields? For example, when the value in PostgreSQL is simply `interval '1' day`, do we receive `P1D` or `P0Y0M1DT0H0M0S`?

Avoid `unwrap()` after `parse()`. We cannot trust the input to be fully valid.

Reply:
Yes. For example, an interval of hour (`01:00:00`) will become `P0Y0M0DT1H0M0S`